### ISSUE TITLE
ButtonBarPagerTabStripViewController last selected tab is not properly deselected

### ISSUE THREAD

**Issue by jdev7** [xmartlabs/XLPagerTabStrip#343]
In the `ButtonBarPagerTabStripViewController` if I have scroll in the `buttonBarView` the indicator is not always updated correctly.

If I have the first item selected (1), and I scroll to the last item, select it(2) and scroll back to the first item(3), this first item doesn´t have the `selectedBar`, but the color is still the `selectedColor` (yellowGreen in the example, rather than grayDark).
But if I scroll again to the end, and back to the beginning(4), the button now is again updated (there´s some sort of `refreshing-the-view-problem`.

Here is how I´m updating the buttons:
```
changeCurrentIndex = { (oldCell: ButtonBarViewCell?, newCell: ButtonBarViewCell?, animated: Bool) -> Void in
    oldCell?.label.textColor = UIColor.grayDark
    newCell?.label.textColor = UIColor.yellowGreen
}
```

1) ![image](https://cloud.githubusercontent.com/assets/10078329/24366985/395cd2b2-131b-11e7-9117-633a0f019744.png)
2) ![image](https://cloud.githubusercontent.com/assets/10078329/24366993/443d29b6-131b-11e7-9cb3-cf6c14b7e7c7.png)
3) ![image](https://cloud.githubusercontent.com/assets/10078329/24367002/4ef31b9a-131b-11e7-898c-0030bbe73710.png)
4) ![image](https://cloud.githubusercontent.com/assets/10078329/24367127/c2b4466c-131b-11e7-8605-ca191b648d23.png)

This is also happening in the provided example:
1) ![image](https://cloud.githubusercontent.com/assets/10078329/24367077/9db9b162-131b-11e7-8e5e-a1fc875eeb47.png)
2) ![image](https://cloud.githubusercontent.com/assets/10078329/24367090/a8b1638a-131b-11e7-9a44-cf3b948e15f8.png)
3) ![image](https://cloud.githubusercontent.com/assets/10078329/24367102/aed6153a-131b-11e7-8cc8-3898bd510901.png)
4) ![image](https://cloud.githubusercontent.com/assets/10078329/24367112/b6edd956-131b-11e7-81f3-1ebdeb447ab8.png)

**Comment by ivanmih** [comment]
same as #338

**Comment by xuzhengrong1** [comment]
the cell offscreen is not  reuse

**Comment by m-revetria** [comment]
This issue was solved in PR #359.
